_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -2067,29 +2082,38 @@ void ImplementationAvc::BrcPreEnc(
 
     m_brc.PreEnc(task.m_brcFrameParams, m_tmpVmeData);
 }
-#if defined(MXF_ENABLE_MCTF_IN_AVC)
-mfxStatus ImplementationAvc::SubmitToMctf(DdiTask * pTask, bool isSceneChange, bool doIntraFiltering)
+#if defined(MFX_ENABLE_MCTF_IN_AVC)
+mfxStatus ImplementationAvc::SubmitToMctf(DdiTask * pTask)
 {
     MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_API, "VideoVPPHW::SubmitToMctf");
-    //mfxStatus sts = MFX_ERR_NONE;
-
-    pTask->m_bFrameReady   = false;
-    mfxFrameSurface1
-        * pSurfI         = nullptr;
-    pSurfI               = m_core->GetNativeSurface(pTask->m_yuv);
-    pSurfI               = pSurfI ? pSurfI : pTask->m_yuv;
 
-    amtMctf->IntBufferUpdate(isSceneChange);
+    pTask->m_bFrameReady = false;
+    bool isIntraFrame    = (pTask->GetFrameType() & MFX_FRAMETYPE_I) || (pTask->GetFrameType() & MFX_FRAMETYPE_IDR);
```

Comment:
pTask->GetFrameType() & (MFX_FRAMETYPE_I | MFX_FRAMETYPE_IDR); would be a bit shorter

Reply:
Definitely, changing that